src/backend/distributed/shardsplit/shardsplit_shared_memory.c

```diff
@@ -0,0 +1,389 @@
+/*-------------------------------------------------------------------------
+ *
+ * shardsplit_shared_memory.c
+ *    API's for creating and accessing shared memory segments to store
+ *    shard split information. 'setup_shard_replication' UDF creates the
+ *    shared memory, populates the contents and WAL sender processes are
+ *    the consumers.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "distributed/shardinterval_utils.h"
+#include "distributed/shardsplit_shared_memory.h"
+#include "distributed/citus_safe_lib.h"
+#include "distributed/multi_logical_replication.h"
+#include "storage/ipc.h"
+#include "utils/memutils.h"
+#include "common/hashfn.h"
+
+const char *SharedMemoryNameForHandleManagement =
+	"Shared memory handle for shard split";
+
+static shmem_startup_hook_type prev_shmem_startup_hook = NULL;
```

Comment:
Did not get a chance to review this last time. For my understanding, can you explain what this 'prev_shmem_startup_hook' is for? A short comment would be nice to as it is not obvious.

Reply:
PG provides `shmem_startup_hook`  to allow modules to initialize static shared memory segments during boot.
We cache this original hook in 'prev_shmem_startup_hook'. 
As a part of Citus shared library initialization, we do static shared memory allocation for Backend management, connection management etc and then call the original hook.